cmd/deploy.go

```diff
@@ -0,0 +1,98 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/pierrre/archivefile/zip"
+	"github.com/skuid/skuid/platform"
+	"github.com/spf13/cobra"
+)
+
+var deployCmd = &cobra.Command{
+	Use:   "deploy",
+	Short: "Deploy local Skuid metadata to a Skuid Platform Site.",
+	Long:  "Deploy Skuid metadata stored within a local file system directory to a Skuid Platform Site.",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		api, err := platform.Login(
+			host,
+			username,
+			password,
+			apiVersion,
+			verbose,
+		)
+
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(1)
+		}
+
+		var targetDirFriendly string
+
+		// If target directory is not provided,
+		// use the current directory's contents
+		if targetDir == "" {
+			targetDir = "."
+			targetDirFriendly, err = filepath.Abs(filepath.Dir(os.Args[0]))
+			if err != nil {
+				log.Fatal(err)
+			}
+		} else {
+			targetDirFriendly = targetDir
+		}
+
+		if verbose {
+			fmt.Println("Deploying site from", targetDirFriendly)
+		}
+
+		// Create a temporary directory into which to put our ZIP
+		tmpDir, err := ioutil.TempDir("", "skuid-deploy")
+		if err != nil {
+			panic(err)
```

Comment:
We probably shouldn't panic here as the user will have no idea what's _actually_ wrong. Maybe log a human parsable error, and exit early.